mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1578,3 +1579,345 @@ bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
+                {
+                    if (m_bOnlyCheckVelocity)
+                    {
+                        m_OnCatapulted.FireOutput(pOther, this);
+                        return;
+                    }
+                    bLaunch = true;
+                }
+            }
+            else
+            {
+                Vector vecLaunchDir = vec3_origin;
+                AngleVectors(m_vLaunchDirection, &vecLaunchDir);
+                float flPlayerEntryAng = DotProduct(vecLaunchDir.Normalized(), vecPlayerVelocity.Normalized());
+
+                // VDC uses brackets so inclusive??
+                if (flPlayerEntryAng >= m_flEntryAngleTolerance)
+                {
+                    if (m_bOnlyCheckVelocity)
+                    {
+                        m_OnCatapulted.FireOutput(pOther, this);
+                        return;
+                    }
+                    bLaunch = true;
+                }
+            }
+        }
+    }
+
+    if (bLaunch)
+    {
+        if (m_bUseLaunchTarget)
+        {
+            LaunchAtTarget(pOther);
+        }
+        else
+        {
+            LaunchAtDirection(pOther);
+        }
+    }
+}
+
+void CTriggerMomentumCatapult::OnStartTouch(CBaseEntity* pOther)
+{
+    BaseClass::OnStartTouch(pOther);
+
+    // Ignore vphys only allow players
+    if (pOther && pOther->IsPlayer())
+    {
+        Launch(pOther);
+
+        if (m_bOnThink)
+            SetNextThink(gpGlobals->curtime + m_flInterval);
+    }
+}
+
+void CTriggerMomentumCatapult::Think()
+{
+    if (m_bOnThink)
+    {
+        FOR_EACH_VEC(m_hTouchingEntities, i)
+        {
+            const auto pEnt = m_hTouchingEntities[i].Get();
+            if (pEnt && pEnt->IsPlayer())
+            {
+                Launch(pEnt);
+                SetNextThink(gpGlobals->curtime + m_flInterval);
+            }
+        }
+    }
+    else
+    {
+        SetNextThink(TICK_NEVER_THINK);
+    }
+}
+
+int CTriggerMomentumCatapult::DrawDebugTextOverlays()
+{
+    int text_offset = BaseClass::DrawDebugTextOverlays();
+
+
+    char tempstr[255];
+
+    Q_snprintf(tempstr, sizeof(tempstr), "Name: %s", GetDebugName());
+    EntityText(text_offset, tempstr, 0);
+    text_offset++;
+
+    Q_snprintf(tempstr, sizeof(tempstr), "Position: %f, %f, %f", GetAbsOrigin().x, GetAbsOrigin().y, GetAbsOrigin().z);
+    EntityText(text_offset, tempstr, 0);
+    text_offset++;
+
+    if (m_target != NULL_STRING)
+    {
+        Q_snprintf(tempstr, sizeof(tempstr), "Launch target: %s", m_target.ToCStr());
+        EntityText(text_offset, tempstr, 0);
+        text_offset++;
+    }
+
+
+    Q_snprintf(tempstr, sizeof(tempstr), "Player velocity: %f", m_flPlayerSpeed);
+    EntityText(text_offset, tempstr, 0);
+    text_offset++;
+
+    Vector vecLaunchVelocity = vec3_origin;
+    if (m_target != NULL_STRING)
+    {
+        // Create dummy velocity
+        Vector vecPlayerOrigin = this->GetAbsOrigin();
```

Comment:
`this->` is redundant.
```suggestion
        Vector vecPlayerOrigin = GetAbsOrigin();
```